app/processing/funding/get-fundings.js

```diff
@@ -1,21 +1,11 @@
-const schema = require('./fundings-schema')
 const mapFundings = require('./map-fundings')
-
+const validateFundings = require('./validate-fundings')
 const getFundingsByCalculationId = require('./get-fundings-by-calculation-id')
 
 const getFundings = async (calculationId, transaction) => {
   const rawFundings = await getFundingsByCalculationId(calculationId, transaction)
   const fundings = await mapFundings(rawFundings)
-
-  const result = schema.validate(fundings, {
-    abortEarly: false
-  })
-
-  if (result.error) {
-    throw new Error(`Calculation with calculationId: ${calculationId} does not have valid funding(s): ${result.error.message}`)
-  }
-
-  return fundings
+  return validateFundings(fundings, calculationId)

```

Comment:
In some of these `map` is wrapped in `validate` call but others they're separate.  Suggest standardising to be consistent.

eg this is in calculation
![image](https://user-images.githubusercontent.com/42574977/199046508-0e352e6d-e61c-49c5-8b3a-36b106d5f612.png)

We're also `mapFundings` is incorrectly declared as async when it does nothing async so can remove that and the await here.

Reply:
The call order of validation and mapping of fundings is not consistent with others because, fundings raw data from the database need to be manipulated (split) for extraction of more details like `name` and `level` before being validated. The validation of some fields depends on another, the dependency value can only be available after mapping.